verticapy/tests_new/performance/vertica/test_qprof.py

```diff
@@ -0,0 +1,1384 @@
+    @pytest.mark.parametrize(
+        "category_order",
+        [
+            # "trace",  # need to implement
+            "category ascending",
+            "category descending",
+            "total ascending",
+            "total descending",
+            # "min ascending",  # need to implement
+            # "min descending",  # need to implement
+            # "max ascending",  # need to implement
+            # "max descending",  # need to implement
+            "sum ascending",
+            "sum descending",
+            # "mean ascending",  # need to implement
+            # "mean descending",  # need to implement
+            # "median ascending",  # need to implement
+            # "median descending",  # need to implement
+        ],
+    )
+    @pytest.mark.parametrize(
+        "kind",
+        [
+            "bar",
+            "barh",
+        ],
+    )
+    @pytest.mark.parametrize(
+        "unit",
+        [
+            "s",
+            "m",
+            "h",
+        ],
+    )
+    # code always sorts on sub_step desc for show=false
+    def test_get_qsteps(self, unit, kind, category_order, show):
+        """
+        test function for get_qsteps
+        """
+
+        qprof = QueryProfiler(transactions=QPROF_SQL2)
+        qprof_steps = qprof.get_qsteps(
+            unit=unit, kind=kind, categoryorder=category_order, show=show
+        )
+
+        # ************************* expected result ************************
+        # logic to get step, sub-step and elapsed_time(in s,m and h)
+        transaction_id, statement_id = qprof.transactions[0]
+        expected_qplans_pdf = vDataFrame(
+            f"select completion_time, time, execution_step from v_internal.dc_query_executions where transaction_id={transaction_id} and statement_id={statement_id}"
+        ).to_pandas()
+
+        # step and sub_step
+        expected_qplans_pdf[["step", "_sub_step"]] = (
+            expected_qplans_pdf["execution_step"]
+            .str.split(":", expand=True)
+            .rename({0: "step", 1: "_sub_step"}, axis=1)
+        )
+        expected_qplans_pdf["sub_step"] = expected_qplans_pdf.apply(
+            lambda x: x["step"] if pd.isnull(x["_sub_step"]) else x["_sub_step"], axis=1
+        )
+
+        # elapsed_time
+        expected_qplans_pdf["elapsed_time"] = (
+            expected_qplans_pdf["completion_time"] - expected_qplans_pdf["time"]
+        ).dt.total_seconds()
+        if unit == "m":
+            expected_qplans_pdf["elapsed_time"] = round(
+                expected_qplans_pdf["elapsed_time"] / 60, 10
+            )
+        elif unit == "h":
+            expected_qplans_pdf["elapsed_time"] = round(
+                expected_qplans_pdf["elapsed_time"] / 3600, 10
+            )
+        else:
+            expected_qplans_pdf["elapsed_time"] = round(
+                expected_qplans_pdf["elapsed_time"], 10
+            )
+        expected_qplans_pdf = expected_qplans_pdf[["step", "sub_step", "elapsed_time"]]
```

Comment:
I also second that. The tests should generally be small. Such that the comparison is easier to read and interpret. 
if it is not too much work, then it may better to still divide it into multiple things. 
Generally, one test should check ONLY one aspect of a function. 
E.g. one test can check if the correct graphics are plotted i.e. bar or pie. 
This can be called test_get_qsteps_plot

Reply:
I will add comments to each test to make it more readable.
Thanks for suggestion. I can try for future test.